app/views/dashboard/advisers/_advisers_table.html.erb

```diff
@@ -9,18 +9,29 @@
       </tr>
     </thead>
     <tbody>
-      <% advisers.each do |adviser| %>
-        <tr class="t-advisers-table-row">
-          <td class="t-adviser-reference-number"><%= adviser.reference_number %></td>
-          <td class="t-adviser-name"><%= adviser.name %></td>
-          <td>
-            <ul class="l-comma-separated-list">
-              <% adviser.qualifications.each do |qualification| %>
-                <li class="t-adviser-qualification"><%= qualification.name %></li>
-              <% end %>
-            </ul>
+      <% if firm.advisers.present? %>
+        <% firm.advisers.each do |adviser| %>
+          <tr class="t-advisers-table-row">
+            <td class="t-adviser-reference-number"><%= adviser.reference_number %></td>
+            <td class="t-adviser-name"><%= adviser.name %></td>
+            <td>
+              <ul class="l-comma-separated-list">
+                <% adviser.qualifications.each do |qualification| %>
+                  <li class="t-adviser-qualification"><%= qualification.name %></li>
+                <% end %>
+              </ul>
+            </td>
+            <td class="t-adviser-postcode"><%= adviser.postcode %></td>
+          </tr>
+        <% end %>
+      <% else %>
+        <tr>
+          <td class="t-no-advisers-message" colspan="4">
+            <%= t('dashboard.advisers_index.no_advisers_message') %>
+            <%= link_to(t('dashboard.advisers_index.add_adviser_button_full', firm_name: firm.registered_name),
+                        new_dashboard_firm_adviser_path(firm),
+                        class: 't-add-adviser-link')  %>.
```

Comment:
If we push the full stop into the locale then I'll need to duplicate the `dashboard.advisers_index.add_adviser_button_full` as surely we don't want this in the button? @neoeno